code/code/cmd/cmd_deathstroke.cc

```diff
@@ -35,96 +41,115 @@ static int deathstroke(TBeing *caster, TBeing *victim)
     caster->sendTo("You can't deathstroke while mounted!\n\r");
     return FALSE;
   }
-  if (dynamic_cast<TBeing *> (victim->riding)) {
-    caster->sendTo("You can't deathstroke a mounted person!\n\r");
-    return FALSE;
-  }
-  if (victim->isFlying() && (victim->fight() != caster)) {
-    caster->sendTo("You can only deathstroke fliers that are fighting you.\n\r");
-    return FALSE;
-  }
   if (!caster->heldInPrimHand() || 
       !(tw=dynamic_cast<TBaseWeapon *>(caster->heldInPrimHand()))) {
     caster->sendTo("You need to hold a weapon in your primary hand to make this a success.\n\r");
     return FALSE;
   }
```

Comment:
Not a big deal, but if you wanted to you could move these hard-fail checks (lines 15-50) into the `doDeathstroke` function to separate concerns a little more. I _think_ that was how they intended it to work - `doSkillName` functions perform the hard fail checks then call the `skillName` functions that actually contains the logic for the skills themselves. But it seems like most of the previous coders ended up splitting hard-fail checks into both functions fairly randomly.